feature_tests/steps/common.py

```diff
@@ -0,0 +1,91 @@
+from uuid import UUID
+
+import parse
+from behave import register_type
+
+
+@parse.with_pattern(
+    r"\{[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}\}"
+)
+def parse_uuid(value):
+    return UUID(value)
+
+
+def parse_string(value):
+    return str(value)
+
+
+@parse.with_pattern(r"-?[0-9]+")
+def parse_number(value):
+    return int(value)
+
+
+@parse.with_pattern(r"[0-9A-Z]{1,5}")
+def parse_ods_code(value):
+    return value
+
+
+@parse.with_pattern(r"(a|an)")
+def parse_a_or_an(value):
+    return value
+
+
+@parse.with_pattern(r"[a-z0-9_]+(\.[a-z0-9_]+)*")
+def parse_path(value):
+    return value
+
+
+# Register the different datatypes used in the step definitions.
+register_type(
+    UUID=parse_uuid,
+    String=parse_string,
+    Number=parse_number,
+    OdsCode=parse_ods_code,
+    An=parse_a_or_an,
+    Path=parse_path,
+)
+
+
+def catch_errors(func):
+    """
+    Decorator that will catch an exception and record it on the context
+    """
+
+    def inner(context, *args, **kwargs):
+        try:
+            func(context, *args, **kwargs)
+        except Exception as e:
+            context.error = e
+
+    return inner
+
+
+def assert_type_matches(obj, expected):
+    actual = type(obj).__name__
+    assert actual == expected, "Type mismatch: expected {expected} got {actual}"
```

Comment:
there's a slight chance of this giving false positives because using the name will circumvent namespacing.

Could add another parse method for types, so that the name is parsed into the correct type explicitly, and then you can instead check type equalities